### ISSUE TITLE
Build a Fitting Tool

### ISSUE THREAD

**Issue by tholoien** [tholoien/empiriciSN#13]
Build a simple tool to use the trained model to produce realistic SN parameters based on input host parameters (from CatSim). 

We could also have this tool produce light curves; once you have the SALT2 parameters it is simple to generate light curves using SNCosmo.

**Comment by drphilmarshall** [comment]
I think the DES team will definitely want this if they don't have the
capability already. Biswas is already set up to this for LSST.

**Comment by tholoien** [comment]
We now have a fitting tool, but need to confirm what host properties will be given --- do we want it to take in colors, local surface brightnesses, etc., and put the responsibility on the user to calculate those quantities, or do we want it to take, e.g., host magnitudes, fluxes, and effective radii, and have our tool calculate the necessary quantities? I think this probably would depend on what properties are easily measured (for DES) or produced by CatSim (for LSST), and we can do the calculations. Thoughts?

I think leaving the actual light curve simulation for the user to do is the way to go. SN Cosmo is one tool to do this, but there are others, and the key thing we want to do is just keep it simple and produce realistic SALT2 parameters so the user can do what they want with them.

**Comment by drphilmarshall** [comment]
Agreed on LC simulation. I think we need to take host galaxy model
parameters theta (ie mags, sersic parameter, size) as inputs, then draw
possible SN positions x from a GMM model of Pr(x | theta). Note that this
PDF has the SN flux and host SB information marginalized out - in practice
we can condition on theta, draw everything else, and ignore everything
except the SN position.

Then, once we have a host model and an SN position, we can compute the
local SB parameters phi of the host at that SN position. That will then
enable us to draw the SN light curve parameters c, from a GMM model of Pr(c
| x, theta, phi) - so conditioning on x, theta, phi and drawing c.

What do you think?

A good test will be to see whether mock SN drawn this way end up with
parameters (c, x, theta, phi) that agree with the original inputs to the
GMM!

**Comment by tholoien** [comment]
A function to predict a SN position from a subset of host parameters has been implemented, and the user can use this to fit a SN in three steps: draw a position with the empiriciSN tool, calculate the surface brightness, and then draw SN parameters with the empiriciSN tool using all the host parameters. We decided not to include the surface brightness calculation as part of this tool so that different methods of calculating the surface brightness can be used.

See the demo notebook for an example of this functionality.
